### PR TITLE
Home component converted to functional component

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**
issue #23 fixed with functional component. Due to some issues with git i have to close another pull request.
@mohitkyadav Sir please review it.
@iiitv/project-maintainers @AmanRaj1608 
Screenshot (103)

### CODE REVIEW

**Review by AmanRaj1608 — APPROVED** [review]

File: src/components/home/HomePage.js [hunk 1]
```diff
@@ -1,41 +1,71 @@
...
-            <div className="uk-offcanvas-content">
-              <button
-                className="uk-button uk-button-default uk-margin-small-right"
-                type="button"
-                data-uk-toggle="target: #offcanvas-push"
-              >
-                Push
-              </button>
+function HomePage() {
+  const [data, setData] = useState(null);
```
Comment (aashutoshrathi) [inline src/components/home/HomePage.js:5]: We need a separate issue to make implement https://aviaryan.com/blog/react-stale-while-revalidate
@AmanRaj1608

File: src/components/home/HomePage.js [hunk 2]
```diff
@@ -1,41 +1,71 @@
...
-          <div className="uk-width-1-5@s" style={{ background: '#0e0b0bb3', color: 'white' }}>
-            <div className="uk-offcanvas-content">
-              <button
-                className="uk-button uk-button-default uk-margin-small-right"
-                type="button"
-                data-uk-toggle="target: #offcanvas-push"
-              >
-                Push
-              </button>
+function HomePage() {
```
Comment (aashutoshrathi) [inline src/components/home/HomePage.js:4]: Better to use a consistent declaration like 
`const HomePage = () => {`

File: src/components/home/HomePage.js [hunk 3]
```diff
@@ -1,41 +1,71 @@
...
+  const [data, setData] = useState(null);
 
-              <div id="offcanvas-push" data-uk-offcanvas="mode: push; overlay: true">
-                <div className="uk-offcanvas-bar">
-                  <button className="uk-offcanvas-close" type="button" data-uk-close="true"></button>
+  useEffect(() => {
+    fetch('https://cors-anywhere.herokuapp.com/https://api.github.com/repos/iiitv/algos/contents', {
```
Comment (aashutoshrathi) [inline src/components/home/HomePage.js:8]: Can't be called without CORS?

File: src/components/home/HomePage.js [hunk 4]
```diff
@@ -1,41 +1,71 @@
...
+      credentials: 'same-origin',
+      headers: new Headers({
+        'Access-Control-Allow-Origin': '*',
+      }),
+      mode: 'cors',
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        // console.log(data);
+        let val = data.map((item) => {
```
Comment (aashutoshrathi) [inline src/components/home/HomePage.js:19]: this is a use case of `Array.filter`, also use `const`.
Also, I feel we should not keep DOM elements in the state, rather keep API response in state and write a loop in render

File: src/components/home/HomePage.js [hunk 5]
```diff
@@ -1,41 +1,71 @@
...
+        </div>
+        <div className="uk-width-4-5@s">
+          <div style={{ background: 'blue' }} className="uk-padding-small">
+            <p>yomo</p>
           </div>
         </div>
       </div>
-    );
-  }
+      <div>
```
Comment (aashutoshrathi) [inline src/components/home/HomePage.js:64]: For `div` without class replace them with `<>`

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]
Left a few comments.
@AmanRaj1608 Can you help him with the comments you find relevant.

File: src/components/home/HomePage.js (hunk 3 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:8]: No sir I tried it many times..but cors error was there..Even stackoverflow solns were also giving error...only this soln helps me out...so i used this heroku link

File: src/components/home/HomePage.js (hunk 4 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:19]: okkk..sir i will fix it.

File: src/components/home/HomePage.js (hunk 5 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:64]: will do

File: src/components/home/HomePage.js (hunk 2 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:4]: will do

File: src/components/home/HomePage.js (hunk 3 above)
Comment (aashutoshrathi) [reply src/components/home/HomePage.js:8]: @mohitkyadav can help with this, he created v1 of algos-web, so definitely worked with this API.

Comment (mohitkyadav) [reply src/components/home/HomePage.js:8]: take a look at https://github.com/mohitkyadav/algos-ngular/blob/master/src/app/services/content.service.ts#L14
@prathmeshkhandelwal1

Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:8]: Sir i fetched the same url & it was giving cors error.

Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:8]: @mohitkyadav Can you provide any soln?

File: src/components/home/HomePage.js (hunk 1 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:5]: @AmanRaj1608 any update here?

File: src/components/home/HomePage.js (hunk 3 above)
Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:8]: @aashutoshrathi Sir can you help in this issue?

Comment (aashutoshrathi) [reply src/components/home/HomePage.js:8]: I'll look into the code in a while.
cc: @mohitkyadav

Comment (prathmeshkhandelwal1) [reply src/components/home/HomePage.js:8]: okkk Sir

**Comment by aashutoshrathi** [comment]
Are there any issues other than CORS left with this PR?
@mohitkyadav @AmanRaj1608 
If no, we can do it in separate PR.

**Comment by AmanRaj1608** [comment]
@aashutoshrathi No sir, we can merge this and can create 2 issues for 
- fixing CORS 
- implementing stale-while-revalidate

**Comment by prathmeshkhandelwal1** [comment]
Any update on this ?? @iiitv/project-maintainers

**Review by mohitkyadav — APPROVED** [review]

**Pull request merged**